### PR TITLE
fix: adjust DNSBL return code interpretation

### LINKED ISSUE #2889: [BUG] DNSBL error response handling is incorrect
### Miscellaneous first checks

- [X] I checked that all ports are open and not blocked by my ISP / hosting provider.
- [X] I know that SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself. I'm confident my setup is correct.

### Affected Component(s)

Mail can no longer be received

### What happened and when does this occur?

```Markdown
Spamhaus recently changed how they deal with AWS and its DNS servers (see https://www.spamhaus.com/resource-center/if-you-query-spamhaus-projects-dnsbls-via-amazon-web-services-dns-move-to-the-free-data-query-service/).

In effect, every DNSBL request made by DMS on our AWS EC2 instance (and I suppose every existing EC2 instance) got the following response:

Oct 30 14:13:29 mail postfix/smtpd[14922]: NOQUEUE: reject: RCPT from mail-lj1-f182.google.com[209.85.208.182]: 554 5.7.1 Service unavailable; Client host [209.85.208.182] blocked using zen.spamhaus.org; Error: 
open resolver; https://www.spamhaus.org/returnc/pub/obfuscated; from=<obfuscated@gmail.com> to=<obfuscated@obfuscated> proto=ESMTP helo=<mail-lj1-f182.google.com>

The response from DNSBL is 127.255.255.254 (see their blog post) with a resulting 554 error code. 
Unfortunately, the result is that every single incoming mail is blocked, which of course was only noticed after a couple of days of missed emails.

According to Spamhaus, DNSBL should be configured in such a way that it deals correctly with error codes (see also https://www.spamhaus.org/news/article/788/spamhaus-dnsbl-return-codes-technical-update). In their words:

"Some users continue to query Spamhaus blocklists via public resolvers, unaware that this means that our data does not actually protect their mail stream. We have introduced an error code for these users to provide a clear signal that there is an issue, and that the mailserver configuration needs to be updated."
```

### What did you expect to happen?

```Markdown
DMS should handle Spamhaus' "new" error codes correctly and only block mail if the sender is actually on a list, and never block mails because of a configuration error.
```

### How do we replicate the issue?

```Markdown
1. Spin up a AWS EC2
2. Set ENABLE_DNSBL to 1
3. Try to receive any mail
...
```

### DMS version

v11.2.0

### What operating system is DMS running on?

Linux

### Which operating system version?

Debian 10 (buster)

### What instruction set architecture is DMS running on?

x86_64 / AMD64

### What container orchestration tool are you using?

Docker Compose

### docker-compose.yml

```yml
mail:
    image: docker.io/mailserver/docker-mailserver:latest
    # If the FQDN for your mail-server is only two labels (eg: example.com),
    # you can assign this entirely to `hostname` and remove `domainname`.
    hostname: mail
    domainname: obfuscated.com
    container_name: mail
    # More information about the mail-server ports:
    # https://docker-mailserver.github.io/docker-mailserver/edge/config/security/understanding-the-ports/
    # To avoid conflicts with yaml base-60 float, DO NOT remove the quotation marks.
    ports:
      - "25:25"    # SMTP  (explicit TLS => STARTTLS)
      - "143:143"  # IMAP4 (explicit TLS => STARTTLS)
      - "465:465"  # ESMTP (implicit TLS)
      - "587:587"  # ESMTP (explicit TLS => STARTTLS)
      - "993:993"  # IMAP4 (implicit TLS)
    environment:
      LETSENCRYPT_HOST: mail.obfuscated.com
      LETSENCRYPT_RESTART_CONTAINER: "true"
      VIRTUAL_HOST: mail.obfuscated.com
      SSL_TYPE: manual
      SSL_CERT_PATH: /etc/letsencrypt/mail.obfuscated.com.crt
      SSL_KEY_PATH: /etc/letsencrypt/mail.obfuscated.com.key
    volumes:
      - maildata:/var/mail
      - mailstate:/var/mail-state
      - maillogs:/var/log/mail
      - mail-config:/tmp/docker-mailserver/
      - /etc/localtime:/etc/localtime:ro
      - ssl:/etc/letsencrypt:ro
    restart: unless-stopped
    stop_grace_period: 1m
    env_file:
      - ./mail/.env
      - ./mail/mailserver.env
    cap_add:
      - NET_ADMIN
      - SYS_PTRACE
    logging:
      options:
        max-size: "250m"
        max-file: "3"
```

### Relevant log output

```bash
Oct 30 14:13:29 mail postfix/smtpd[14922]: NOQUEUE: reject: RCPT from mail-lj1-f182.google.com[209.85.208.182]: 554 5.7.1 Service unavailable; Client host [209.85.208.182] blocked using zen.spamhaus.org; Error: 
open resolver; https://www.spamhaus.org/returnc/pub/obfuscated; from=<obfuscated@gmail.com> to=<obfuscated@obfuscated> proto=ESMTP helo=<mail-lj1-f182.google.com>
```

### Other relevant information

_No response_

### What level of experience do you have with Docker and mail servers?

- [ ] I am inexperienced with docker
- [ ] I am inexperienced with mail servers
- [ ] I am uncomfortable with the CLI

### Code of conduct

- [X] I have read this project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree
- [X] I have read the [README](https://github.com/docker-mailserver/docker-mailserver/blob/master/README.md) and the [documentation](https://docker-mailserver.github.io/docker-mailserver/edge/) and I searched the [issue tracker](https://github.com/docker-mailserver/docker-mailserver/issues?q=is%3Aissue) but could not find a solution

### Improvements to this form?

```Markdown
"What level of experience do you have with Docker and mail servers?" offers only answers in the negative form. I don't mind indicating that I have experience and am comfortable with the CLI, but I do not have that option. You can only guess why I didn't click the negative options if you do not give me the positive options.
```

### PR DESCRIPTION
**Pull request by georglauterbach**
# Description

Adjusts the DNSBL return codes interpretation. Now, only "valid" return codes, i.e. the ones that indicate spam, are taken into account.

Fixes #2889

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by polarathene — DISMISSED** [review]

File: target/scripts/startup/setup-stack.sh [hunk 1]
```diff
@@ -399,7 +399,7 @@ function _setup_postgrey
   _log 'debug' 'Configuring Postgrey'
 
   sed -i -E \
-    's|, reject_rbl_client zen.spamhaus.org$|, reject_rbl_client zen.spamhaus.org, check_policy_service inet:127.0.0.1:10023|' \
+    's|, reject_rbl_client zen.spamhaus.org=127.0.0.[2..11]$|, reject_rbl_client zen.spamhaus.org=127.0.0.[2..11], check_policy_service inet:127.0.0.1:10023|' \
```
Comment (casperklein) [inline target/scripts/startup/setup-stack.sh:402]: Isn't `[2..11]` interpreted as character class and should be escaped?

File: test/mail_with_postgrey.bats [hunk 2]
```diff
@@ -25,7 +25,7 @@ function teardown_file() {
 }
 
 @test "checking postgrey: /etc/postfix/main.cf correctly edited" {
-  run docker exec mail_with_postgrey /bin/bash -c "grep 'zen.spamhaus.org, check_policy_service inet:127.0.0.1:10023' /etc/postfix/main.cf | wc -l"
+  run docker exec mail_with_postgrey /bin/bash -c "grep 'zen.spamhaus.org=127.0.0.[2..11], check_policy_service inet:127.0.0.1:10023' /etc/postfix/main.cf | wc -l"
```
Comment (casperklein) [inline test/mail_with_postgrey.bats:28]: `grep -F` or escape `[2..11]`.

File: target/scripts/startup/setup-stack.sh [hunk 3]
```diff
@@ -1078,7 +1078,7 @@ function _setup_dnsbl_disable
   _log 'debug' 'Disabling postfix DNS block list (zen.spamhaus.org)'
 
   sedfile -i \
-    '/^smtpd_recipient_restrictions = / s/, reject_rbl_client zen.spamhaus.org//' \
+    '/^smtpd_recipient_restrictions = / s/, reject_rbl_client zen.spamhaus.org=127.0.0.\[2..11\]//' \
```
Comment (casperklein) [inline target/scripts/startup/setup-stack.sh:1081]: Same here.

**Review by casperklein — CHANGES_REQUESTED** [review]

File: target/scripts/startup/setup-stack.sh (hunk 1 above)
Comment (georglauterbach) [reply target/scripts/startup/setup-stack.sh:402]: Probably, yes - thanks :)

File: target/scripts/startup/setup-stack.sh (hunk 3 above)
Comment (georglauterbach) [reply target/scripts/startup/setup-stack.sh:1081]: The current diff already shows it escaped (`\[2..11\]`). Is there more to it?

Comment (casperklein) [reply target/scripts/startup/setup-stack.sh:1081]: My fault.

File: target/scripts/startup/setup-stack.sh [hunk 4]
```diff
@@ -398,8 +398,8 @@ function _setup_postgrey
 {
   _log 'debug' 'Configuring Postgrey'
 
-  sed -i -E \
-    's|, reject_rbl_client zen.spamhaus.org$|, reject_rbl_client zen.spamhaus.org, check_policy_service inet:127.0.0.1:10023|' \
+  sedfile -i -E \
+    's|(smtpd_recipient_restrictions.*)|\1, check_policy_service inet:127.0.0.1:10023|' \
```
Comment (casperklein) [inline target/scripts/startup/setup-stack.sh:402]: I like this. Even more future safe:

```suggestion
    's|(^smtpd_recipient_restrictions =.*)|\1, check_policy_service inet:127.0.0.1:10023|' \
```

**Review by casperklein — APPROVED** [review]

**Review by polarathene — APPROVED** [review]

**Pull request merged**